bindings/java/src/main/com/apple/foundationdb/EventKeeper.java

```diff
@@ -0,0 +1,172 @@
+	 */
+	void count(Event event, long amt);
+
+	/**
+	 * Convenience method to add 1 to the number of events which occurred.
+	 *
+	 * @param event the event which occurred.
+	 */
+	default void increment(Event event) { count(event, 1L); }
+
+	/**
+	 * Count the time taken to perform an event, in nanoseconds.
+	 *
+	 * Note that {@code event.isTimeEvent()} should return true here.
+	 *
+	 * @param event the event which was timed (the event should be a time event).
+	 * @param nanos the amount of time taken (in nanoseconds)
+	 */
+	void timeNanos(Event event, long nanos);
+
+	/**
+	 * Count the time taken to perform an action, in the specified units.
+	 *
+	 * Note that {@code event.isTimeEvent()} should return true.
+	 *
+	 * @param event the event which was timed.
+	 * @param duration the time taken
+	 * @param theUnit the unit of time in which the time measurement was taken
+	 */
+	default void time(Event event, long duration, TimeUnit theUnit) { timeNanos(event, theUnit.toNanos(duration)); }
+
+	/**
+	 * Get the number of events which occurred since this timer was created.
+	 *
+	 * If the event was never recorded, then this returns 0.
+	 *
+	 * @param event the event to get the count for
+	 * @return the number of times the event was triggered. If the event has never been triggered,
+	 * then this returns 0
+	 */
+	long getCount(Event event);
+
+	/**
+	 * Get the amount of time taken by this event, in nanoseconds.
+	 *
+	 * @param event the event to get the time for
+	 * @return  the total time measured for this event, in nanoseconds. If the event was never recorded,
+	 * return 0 instead.
+	 */
+	long getTimeNanos(Event event);
+
+	/**
+	 * Get the amount of time taken by this event, in the specified units.
+	 *
+	 * Important note: If the time that was measured in nanoseconds does not evenly divide the unit that
+	 * is specified (which is likely, considering time), then some precision may be lost in the conversion. Use
+	 * this carefully.
+	 *
+	 * @param event the event to get the time for
+	 * @param theUnit the unit to get time in
+	 * @return  the total time measured for this event, in the specified unit. If the event was never recorded,
+	 * return 0.
+	 */
+	default long getTime(Event event, TimeUnit theUnit) {
+		return theUnit.convert(getTimeNanos(event), TimeUnit.NANOSECONDS);
+	}
+
+	/**
+	 * Marker interface for tracking the specific type of event that occurs, and metadata about said event.
+	 *
+	 * Implementations should be sure to provide a quality {@code equals} and {@code hashCode}.
+	 */
+	interface Event {
+		/**
+		 * @return the name of this event, as a unique string. This name should generally be unique, because
+		 * it's likely that {@code EventKeeper} implementations will rely on this for uniqueness.
+		 */
+		String name();
+
+		/**
+		 * @return true if this event represents a timed event, rather than a counter event.
+		 */
+		default boolean isTimeEvent() { return false; };
```

Comment:
I don't know how often this interface is going to be implemented outside of the core FDB bindings logic, so I'm not sure that it's a huge deal either way

Reply:
It's probably not a huge deal either way, no, I'm mostly just commenting because it may be something we want people (including us within the FDB bindings) to be encouraged to think about and be explicit about rather than (accidentally) relying on the default value. But it's also probably fine as is